src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/AbstractMemberScopedReferenceFinder.cs

```diff
@@ -114,81 +106,58 @@ protected sealed override bool CanFind(TSymbol symbol)
 
         protected static ValueTask<ImmutableArray<FinderLocation>> FindReferencesInTokensWithSymbolNameAsync(
             TSymbol symbol,
-            Document document,
-            SemanticModel semanticModel,
-            FindReferenceCache cache,
+            FindReferencesDocumentState state,
             IEnumerable<SyntaxToken> tokens,
             CancellationToken cancellationToken)
         {
             return FindReferencesInTokensWithSymbolNameAsync(
-                symbol, document, semanticModel, cache, tokens, findParentNode: null, cancellationToken);
+                symbol, state, tokens, findParentNode: null, cancellationToken);
         }
 
         protected static ValueTask<ImmutableArray<FinderLocation>> FindReferencesInTokensWithSymbolNameAsync(
             TSymbol symbol,
-            Document document,
-            SemanticModel semanticModel,
-            FindReferenceCache cache,
+            FindReferencesDocumentState state,
             IEnumerable<SyntaxToken> tokens,
-            Func<SyntaxToken, SyntaxNode>? findParentNode,
+            Func<FindReferencesDocumentState, SyntaxToken, SyntaxNode>? findParentNode,
             CancellationToken cancellationToken)
         {
-            var name = symbol.Name;
-            var syntaxFacts = document.GetRequiredLanguageService<ISyntaxFactsService>();
-            var symbolsMatch = GetStandardSymbolsMatchFunction(
-                symbol, findParentNode, document.Project.Solution, cache, cancellationToken);
-
             return FindReferencesInTokensAsync(
-                document,
-                semanticModel,
-                cache,
+                state,
                 tokens,
-                t => IdentifiersMatch(syntaxFacts, name, t),
-                symbolsMatch,
+                static (state, token, name, _) => IdentifiersMatch(state.SyntaxFacts, name, token),
```

Comment:
many lambdas became static by passing along the state object so that data from it could be used instead of captured.